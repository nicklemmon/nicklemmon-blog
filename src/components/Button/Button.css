@import '../../variables/variables.css';

.Button {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: var(--spacingMD) var(--spacingLG);
  font-family: var(--fontFamilySecondary);
  border-radius: var(--borderRadius);
  border: 2px solid #000; /* black is just a placeholder to be overidden */
  font-weight: 700;
  font-size: 1rem;
  cursor: pointer;
  transition: background-color var(--duration) var(--easing);
}

.Button--full {
  width: 100%;
}

.Button--centered {
  margin: 0 auto;
}

.Button--primary {
  background-color: var(--blueRibbon);
  color: var(--white);
  border-color: var(--blueRibbon);
}

.Button--primary:hover {
  background-color: var(--blue);
}

.Button--primary:focus {
  outline-color: var(--blue);
}

.Button--secondary {
  background-color: var(--white);
  color: var(--blueRibbon);
  border-color: var(--blueRibbon);
}

.Button--tertiary {
  background-color: var(--white);
  color: var(--blueRibbon);
  border-color: transparent;
}

.Button--secondary:hover,
.Button--tertiary:hover {
  background-color: var(--seaShell);
}

.Button--ghost {
  background-color: transparent;
  color: var(--white);
  border-color: var(--white);
}

.Button-icon--pre {
  margin-right: var(--spacingSM);
}

.Button-icon--post {
  margin-left: var(--spacingSM);
}
