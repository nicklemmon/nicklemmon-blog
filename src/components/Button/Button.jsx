import React from 'react'
import { Link } from 'gatsby'
import classNames from 'classnames'

import './Button.css';

export default class Button extends React.Component {
  render() {
    const {
      to,
      type,
      children,
      fullWidth,
      centered,
      className
    } = this.props

    return (
      <Link
        to={ to }
        className={ classNames( `Button Button--${type}`, fullWidth ? 'Button--full' : '', centered ? 'Button--centered' : '', className ) }
      >
        { children }
      </Link>
    )
  }
}