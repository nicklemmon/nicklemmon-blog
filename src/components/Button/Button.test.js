import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Button from './Button'

test( 'Button', async () => {
  const { getByText } = render( <Button to="/">Click Me</Button> )

  expect( getByText( 'Click Me' ) ).not.toBeNull()
} )
