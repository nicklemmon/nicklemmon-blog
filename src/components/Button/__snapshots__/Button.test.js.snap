// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button href correctly 1`] = `
<a
  className="Button Button--primary"
  href="/about"
>
  About
</a>
`;

exports[`renders button type correctly 1`] = `
<a
  className="Button Button--primary"
>
  About
</a>
`;
